File: engine/operations/verification.py
"""Operation 探针入口；具体 CLI 执行由 adapter verifier 持有。"""

from ..context import EngineContext
from ..errors import require_agent_capability


class ProbeTimeout(RuntimeError):
    pass


def timeout_report(tool, error) -> dict:
    return {"status": "failed", "code": "probe.timeout",
            "params": {"tool": tool},
            "diagnostic": {"stdout": "", "stderr": str(error),
                           "truncated": False}}


def run_probe(tool, session_id, dirpath=None, model=None, *,
              ports: EngineContext):
    try:
        verifier = require_agent_capability(
            ports.adapter(tool), "probe", "verifier",
        )
        return verifier.probe(session_id, dirpath, model)
    except Exception as error:
        if error.__class__.__name__ == "ProbeTimeout":
            raise ProbeTimeout(str(error)) from error
        raise

File: crates/ferry-engine/src/operations/verification.rs
//! Operation agent prompt 入口；具体 CLI 执行由 adapter verifier 持有。

use serde_json::{Map, Value};

use crate::operations::types::{EngineResult, Ports};

/// 等价 `verification.run_agent_prompt`；`timeout` 默认 360 秒由分发层给。
pub fn run_agent_prompt(
    tool: &str,
    session_id: &str,
    prompt: &str,
    dirpath: Option<&str>,
    model: Option<&str>,
    timeout: u64,
    ports: &Ports,
) -> EngineResult<Map<String, Value>> {
    let adapter = ports.adapter(tool)?;
    let verifier = adapter.require_verifier("prompt")?;
    Ok(verifier.prompt_session(session_id, dirpath, prompt, model, timeout)?)
}
